cluster/defaults.go

```diff
@@ -1022,11 +1023,25 @@ func (c *Cluster) setCloudProvider() error {
 	if p != nil {
 		c.CloudConfigFile, err = p.GenerateCloudConfigFile()
 		if err != nil {
-			return fmt.Errorf("Failed to parse cloud config file: %v", err)
+			return fmt.Errorf("failed to parse cloud config file: %v", err)
 		}
 		c.CloudProvider.Name = p.GetName()
 		if c.CloudProvider.Name == "" {
-			return fmt.Errorf("Name of the cloud provider is not defined for custom provider")
+			return fmt.Errorf("name of the cloud provider is not defined for custom provider")
+		}
+		if c.CloudProvider.Name == aws.AWSCloudProviderName {
```

Comment:
Is `external-aws` valid when the cluster version is < 1.27? Do we need to add a check similar to the existing one?

Reply:
Yeah, it's valid as out-of-tree is supported for <1.27 but wasn't mandatory. There could be use cases where users might migrate to external first before upgrading to 1.27 so only validating for aws in-tree.

Reply:
Right, it makes sense to allow it for migration.